File: dojo/si.j/FizzBuzz/CSharp/FizzBuzzKata/FizzBuzz.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace FizzBuzzKata
{
    public class FizzBuzz
    {
        public String FizzOrBuzz(int n) {
            StringBuilder sb = new System.Text.StringBuilder("");

            if (n % 3 == 0) { 
                sb.Append("Fizz"); 
            }
            if (n % 5 == 0) { 
                sb.Append("Buzz"); 
            }
            if (!(n % 3 == 0) && !(n % 5 == 0)) {
                sb.Append(n.ToString());
            }
            return sb.ToString();
        }

        public String FizzOrBuzz2(int n) {
            StringBuilder sb = new System.Text.StringBuilder("");
            String num = n.ToString();
            bool fizzFlag = false;
            bool buzzFlag = false;

            if (n % 3 == 0 || num.Contains("3")) {
                fizzFlag = true;
                sb.Append("Fizz");
            }
            if (n % 5 == 0 || num.Contains("5")) {
                buzzFlag = true;
                sb.Append("Buzz");
            }
            if (!fizzFlag && !buzzFlag) {
                sb.Append(n.ToString());
            }
            return sb.ToString();
        }
    }
}

File: dojo/sc.b/FizzBuzz/Ruby/2-15-2013 White Belt/FizzBuzz/fizzbuzz.rb
class FizzBuzz
  def print(numbers = 1..15)
    numbers.map do |num, fizz, buzz|
      case [isFizz(num),isBuzz(num)]
        when [true,true];"FizzBuzz"
        when [true,false];"Fizz"
        when [false,true];"Buzz"
        else num
      end
    end
  end
  def isFizz(num)
    num % 3 == 0
  end
  def isBuzz(num)
    num % 5 == 0
  end
end